src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -0,0 +1,361 @@
+<?php
+/**
+ * Process the different data sources for site-level
+ * config and offers and API to work with them.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Class that abstracts the processing
+ * of the different data sources.
+ */
+class WP_Theme_JSON_Resolver {
```

Comment:
Is there a way to mark all these introduced classes as "internal" to allow them to refactor/update the code in future versions?

Reply:
I don't know. I've seen some function names prepended as `_function_name` when they're only expected to be used internally. Can we do the same for the classes or using something like `_EXPERIMENTAL_`?

Reply:
Maybe we can just mark it `@access private` or something like that. @desrosj would know maybe?

Reply:
Done that at https://github.com/WordPress/wordpress-develop/pull/1262/commits/665801c91d9c844dca6cdd77e814bde64bc862fe